hydrosdk/model.py

```diff
@@ -561,91 +564,67 @@ def __init__(self, name, id_, contract, version, metadata):
         self.id_ = id_
 
 
-class BuildStatus(Enum):
-    """
-    Model building statuses
-    """
-    BUILDING = "BUILDING"
-    FINISHED = "FINISHED"
-    FAILED = "FAILED"
-
-
 class UploadResponse:
     """
     Received status from server about uploading
     """
+
     def __init__(self, model, version_id):
         self.cluster = model.cluster
         self.model = model
         self.model_version_id = version_id
         self.cluster = self.model.cluster
-        self._logs_iterator = self.logs()
-        self.last_log = ""
-        self._status = ""
 
     def logs(self):
         """
         Sends request, saves and returns logs iterator
         :return: log iterator
         """
-        logger = logging.getLogger("ModelDeploy")
-        try:
-            url = "/api/v2/model/version/{}/logs".format(self.model_version_id)
-            logs_response = self.model.cluster.request("GET", url, stream=True)
-            self._logs_iterator = sseclient.SSEClient(logs_response).events()
-        except RuntimeError:
-            logger.exception("Unable to get build logs")
-            self._logs_iterator = None
-        return self._logs_iterator
+        url = "/api/v2/model/version/{}/logs".format(self.model_version_id)
+        logs_response = self.model.cluster.request("GET", url, stream=True)
+        return sseclient.SSEClient(logs_response).events()
 
-    def set_status(self) -> None:
+    def poll_model(self) -> None:
         """
         Checks last log record and sets upload status
         :raises StopIteration: If something went wrong with iteration over logs
         :return: None
         """
-        try:
-            if self.last_log.startswith("Successfully tagged"):
-                self._status = BuildStatus.FINISHED
-            else:
-                self.last_log = next(self._logs_iterator).data
-                self._status = BuildStatus.BUILDING
-        except StopIteration:
-            if not self._status == BuildStatus.FINISHED:
-                self._status = BuildStatus.FAILED
+        model = Model.find(self.cluster, self.model.name, self.model.version)
+        if model:
+            self.model = model
+        else:
+            raise ValueError(
```

Comment:
There is an exception handling inside Model.find

Reply:
my bad didn't notice